internal/suggestions/suggest.go

```diff
@@ -0,0 +1,218 @@
+
+func StartSuggest(ctx context.Context, schemaPath string, suggestionsConfig *Config, outputHints bool) error {
+	fmt.Println("Validating OpenAPI spec...")
+	fmt.Println()
+
+	schema, err := os.ReadFile(schemaPath)
+	if err != nil {
+		return fmt.Errorf("failed to read schema file %s: %w", schemaPath, err)
+	}
+
+	schema, err = ReformatFile(schema, DetectFileType(schemaPath))
+	if err != nil {
+		return fmt.Errorf("failed to reformat schema file %s: %w", schemaPath, err)
+	}
+
+	vErrs, vWarns, vInfo, err := validation.Validate(schema, schemaPath, outputHints)
+	if err != nil {
+		return err
+	}
+
+	printValidationSummary(vErrs, vWarns, vInfo)
+
+	toSuggestFor := append(vErrs, vWarns...) // TODO append(append(vErrs, vWarns...), vInfo...)
+
+	if len(toSuggestFor) > 0 {
+		err = Suggest(schema, schemaPath, toSuggestFor, *suggestionsConfig)
+		if err != nil {
+			fmt.Println(promptui.Styler(promptui.FGRed, promptui.FGBold)(fmt.Sprintf("cannot fetch llm suggestions: %s", err.Error())))
+			return err
+		}
+	} else {
+		fmt.Println(promptui.Styler(promptui.FGGreen, promptui.FGBold)("Congrats! 🎊 Your spec had no issues we could detect."))
+	}
+
+	return nil
+}
+
+func Suggest(schema []byte, schemaPath string, errs []error, config Config) error {
+	suggestionToken := ""
+	fileType := ""
+	totalSuggestions := 0
+
+	l := log.NewLogger(schemaPath)
+
+	// local authentication should just be set in env variable
+	if os.Getenv("SPEAKEASY_SERVER_URL") != "http://localhost:35290" {
+		if err := auth.Authenticate(false); err != nil {
+			return err
+		}
+	}
+
+	if _, err := GetOpenAIKey(); err != nil {
+		return err
+	}
+
+	suggestionToken, fileType, err := Upload(schema, schemaPath)
+	if err != nil {
+		return err
+	} else {
+		// Cleanup Memory Usage in LLM
+		defer func() {
+			Clear(suggestionToken)
+		}()
+
+		// Handle Signal Exit
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+		go func() {
+			<-c
+			Clear(suggestionToken)
+			os.Exit(0)
+		}()
+	}
+
+	suggest, err := New(suggestionToken, schemaPath, fileType, config.Model, schema, config)
+	if err != nil {
+		return err
+	}
+
+	/**
+	 * Parallelized suggestions
+	 */
+	if config.Parallelize {
+		fmt.Println("Getting suggestions...")
+		fmt.Println()
+
+		suggest.Verbose = false
+
+		res, err := suggest.FindSuggestions(errs)
+		if err != nil {
+			return err
+		}
+
+		for i, err := range errs {
+			suggestion := res[i]
+
+			printVErr(l, err)
+			fmt.Println() // Spacing
+			suggestion.Print()
+			fmt.Println(promptui.Styler(promptui.FGGreen, promptui.FGBold)("✓ Suggestion is valid and resolves the error"))
+			fmt.Println() // Spacing
+
+			if suggestion != nil && suggest.AwaitShouldApply() {
+				newFile, err := suggest.ApplySuggestion(*suggestion)
+				if err != nil {
+					return err
+				}
+
+				err = suggest.CommitSuggestion(newFile)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		return nil
+	}
+
+	/**
+	 * Non-parallelized suggestions
+	 */
```

Comment:
I imagine we'll just remove this soon. Don't think there's a reason to not parallelize

Reply:
Agree, let's just leave it in the interim until we have confidence in parallelization